File: PeriodisationProgramApp.Web/src/types/enitities/CommunityEntity.ts
import { BaseEntity } from "./BaseEntity";
import { User } from "./User";

export type CommunityEntity = {
  rating: number;
  userRating: number | null;
  likes: number;
  user: User;
  isLiked: boolean;
  isRated: boolean;
  isPublic: boolean;
} & BaseEntity;

File: PeriodisationProgramApp.Domain/Entities/CommunityEntity.cs
using PeriodisationProgramApp.Domain.Interfaces;

namespace PeriodisationProgramApp.Domain.Entities
{
    public class CommunityEntity<UserLike, UserRating> : BaseEntity, ICommunityEntity<UserLike, UserRating>
                                                                                         where UserLike : IUserLike
                                                                                         where UserRating : IUserRating
    {
        public Guid UserId { get; set; }

        public User? User { get; set; }

        public bool IsPublic { get; set; }

        public int Likes { get; set; }

        public int Rates { get; set; }

        public double Rating { get; set; }

        public List<UserLike> UserLikes { get; set; } = new();

        public List<UserRating> UserRatings { get; set; } = new();
    }
}
